### PR TITLE
GPII-2865: Implement Network policies

### PR DESCRIPTION
**Pull request by stepanstipl**
This PR implements Network Policies for services in GPII namespace - [GPII-2865](https://issues.gpii.net/browse/GPII-2865).

This is the first part, which "only" implements allow policies, but not default "deny", and will be followed by #486 PR which implements default deny. This is in order to prevent any downtime during rollout.

This PR:
- Adds following Ingress Network policies:
  - For CouchDB - allowing all traffic between nodes on ports 5984, 4369 and 9100
  - For Dataloader - allowing access to CouchDB on port 5984
  - For Flushtokens - allowing access to CouchDB on port 5984
  - For Preferences - allowing access to CouchDB on port 5984
  - For Flowmanager
    - allowing access to CouchDB on port 5984
    - allowing access to Preferences on http port 8081
    - allowing access from Istio Ingress gateway to http port 8081
  - For CouchDB Prometheus Exporter - allowing access to CouchDB on port 5984
  - For Locust - optionally allow access to Preferences
- Minor format refactor of some related Helm values (ie. `couchdbUri` -> `couchdb.uri`)

### CODE REVIEW

**Comment by amatas** [comment]
Also, please add the nº of the ticket to the title of this PR to make easier the search in the future.

**Comment by stepanstipl** [comment]
> Also, please add the nº of the ticket to the title of this PR to make easier the search in the future.

Thanks for reviews @amatas, JIRA ticket # is in the first sentence of the description of both PRs (with an actual link), also in the commit message of every single commit. I have also added it to the PR title.

File: shared/charts/couchdb-prometheus-exporter/templates/deployment.yaml [hunk 1]
```diff
@@ -29,13 +29,13 @@ spec:
...
         env:
         - name: COUCHDB_URI
-          value: {{ .Values.couchdbUri }}
+          value: {{ .Values.couchdb.uri }}
         - name: COUCHDB_USERNAME
-          value: {{ .Values.couchdbUsername }}
+          value: {{ .Values.couchdb.username }}
         - name: COUCHDB_PASSWORD
-          value: {{ .Values.couchdbPassword }}
+          value: {{ .Values.couchdb.password }}
```
Comment (natarajaya) [inline shared/charts/couchdb-prometheus-exporter/templates/deployment.yaml]: Looks like this should be added to GPII-4074 and also populated via secret reference.

File: shared/charts/couchdb/templates/network_policy.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      app: {{ template "couchdb.name" . }}
+      release: {{ .Release.Name }}
+  policyTypes:
+  - Ingress
+  ingress:
+  - from:
+    - podSelector:
+        matchLabels:
+          app: {{ template "couchdb.name" . }}
+          release: {{ .Release.Name }}
```
Comment (natarajaya) [inline shared/charts/couchdb/templates/network_policy.yaml]: This policy can be made more strict by limiting ingress to only CouchDB ports: 5984, 4369 and 9100.

**Comment by natarajaya** [comment]
LGTM

File: shared/charts/couchdb-prometheus-exporter/templates/deployment.yaml (hunk 1 above)
Comment (stepanstipl) [reply shared/charts/couchdb-prometheus-exporter/templates/deployment.yaml]: I have added this and other affected workloads to the [GPII-4074](https://issues.gpii.net/browse/GPII-4074) ticket description.

File: shared/charts/couchdb/templates/network_policy.yaml (hunk 2 above)
Comment (stepanstipl) [reply shared/charts/couchdb/templates/network_policy.yaml]: I wasn't sure about the ports, but makes sense to be consistent with the rest of the policies. 👍

**Comment by stepanstipl** [comment]
thanks for review @natarajaya, I have restricted the CouchDB policy to specific orts only. 👍

**Comment by mrtyler** [comment]
> JIRA ticket # is in the first sentence of the description of both PRs (with an actual link), also in the commit message of every single commit. I have also added it to the PR title.

I believe the PR title is the only place where the Jira PR plugin will look for the ticket number: https://issues.gpii.net/browse/GPII-2865?devStatusDetailDialog=pullrequest

File: shared/charts/locust/values.yaml [hunk 3]
```diff
@@ -38,3 +38,7 @@ worker:
     requests:
       cpu: 100m
       memory: 128Mi
+
+target:
+  labels:
+  namespace: gpii
```
Comment (mrtyler) [inline shared/charts/locust/values.yaml]: I think this line should be indented one more level.

File: shared/charts/couchdb-prometheus-exporter/README.md [hunk 4]
```diff
@@ -48,10 +48,12 @@ Parameter | Description | Default
 --- | --- | ---
 `replicaCount` | desired number of pods | `1`
 `exporterListenPort` | port for exporter service to listen on | `9984`
-`couchdbUri` | URI for couchdb | `http://couchdb-svc-couchdb.default.svc.cluster.local:5984`
-`couchdbUsername` | username for couchdb uri | `admin`
-`couchdbPassword` | password for couchdb uri | `password`
-`couchdbDatabases` | list of specific databases to monitor | `_all_dbs`
+`couchdb.labels` | CouchDB pod labels for `NetworkPolicy` | `{ app: "couchdb" }`
+`couchdb.port` | CouchDB port for `NetworkPolicy` | `http-couchdb`
```
Comment (mrtyler) [inline shared/charts/couchdb-prometheus-exporter/README.md]: I understand what this is, but when I see `couchdb.port`, I expect to find a value like `5984`. How do you feel about `portName` or `podPortName` or something? Maybe `port` being a name isn't confusing to others in this Kubernetes context where ports are often names?

File: shared/charts/locust/values.yaml (hunk 3 above)
Comment (stepanstipl) [reply shared/charts/locust/values.yaml]: Done.

File: shared/charts/couchdb-prometheus-exporter/README.md (hunk 4 above)
Comment (stepanstipl) [reply shared/charts/couchdb-prometheus-exporter/README.md]: I kept the name `port` as both forms are accepted here, but improved comment by adding `(can either be a numerical or named port on a pod)` to this and all the other charts.

**Comment by stepanstipl** [comment]
Thanks for review @mrtyler 

> I believe the PR title is the only place where the Jira PR plugin will look for the ticket number: 

https://issues.gpii.net/browse/GPII-4023?devStatusDetailDialog=pullrequest

**Pull request merged**